src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
+                        disableGutters
+                        sx={style.flexSpace.between}
+                    >
+                        <Typography
+                            variant="h6"
+                            noWrap
+                            component="a"
+                            href="/"
+                            sx={{...style.typography, ...style.typography.medium}}
+                        >
+                            DevPrep + LOGO
+                        </Typography>
+                        <Typography
+                            variant="h6"
+                            noWrap
+                            component="a"   
+                            href="/"
+                            sx={{...style.typography, ...style.typography.small}}
+                        >
+                            LOGO
+                        </Typography>
+                        
+                        <SmallView>
+                        <Box>                                
+                            {toggleButton}
+                        </Box>
+                        <IconButton
+                            sx={style.margin.iconBtn}
+                            aria-label="open drawer"
+                            edge="start"
+                            onClick={handleDrawerToggle}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                        </SmallView>
+                        <Responsive>
+                            <Box>
+                                <Button>
+                                    <Link style={Links} to="/posts/general">
+                                        General
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/learn">
+                                        Learn
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/interview">
+                                        Interview
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/project">
+                                        Project
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/login">
+                                        SignIn
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/register">
+                                        SignUp
```

Comment:
It makes sense to keep the register, login and profile links separate from the main nav links. And likely styled differently too. We likely also want to look at a way to style the active nav link too.
It would make sense to make the profile link contain an icon rather than text though this can be done later if necessary